aten/src/ATen/native/cuda/UpSampleBilinear2d.cu

```diff
@@ -13,6 +13,76 @@ namespace at {
 namespace native {
 namespace {
 
+template <
+    typename scalar_t,
+    typename std::enable_if<std::is_same<c10::Half, scalar_t>::value>::type* =
+        nullptr>
+__device__ __forceinline__ void fastSpecializedAtomicAdd(
+    scalar_t* tensor,
```

Comment:
Wait, what would happen if someone decided to play a practical joke on you and send you a tensor that's not half2 aligned (that could happen with e.g. `a=b[1:]`)? You'd need to check alignment of actual pointers.